.github/workflows/continuous-integration.yml

```diff
@@ -68,24 +68,21 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v3
-      - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v4
+      - name: Base Setup
+        uses: jupyterlab/maintainer-tools/.github/actions/base-setup@v1
         with:
-          python-version: ${{ matrix.python-version }}
+          python_version: ${{ matrix.python-version }}
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
-          # All dependencies but markdown-it-py
-          python -m pip install nbformat pyyaml toml
-          python -m pip install -r requirements-dev.txt
           # install sphinx_gallery and matplotlib if available
           python -m pip install sphinx_gallery~=0.7.0 || true
           python -m pip install jupyter-fs || true
       - name: Install markdown-it-py
         if: ${{ matrix.markdown-it-py-version }}
         run: python -m pip install markdown-it-py${{ matrix.markdown-it-py-version }}
```

Comment:
Noted. Usually you can add another `optional-dependency` of `[markdwon]` and then have the action run with and without it (in addition to `[test-cov]`). Currently the `markdown-it-py` is a required dependency, shall wemove it to an optional?

Reply:
Agreed, I'll comment later on the naming convention and organization of `optional-dependency`. @mwouts need your feedback on `markdown-it-py`. Is it meant to be an `optional-dependency` or a hard `dependency`?

Reply:
`markdown-it-py` should remain a hard dependency, because we need it to provide the MyST format, which is the probably the most popular markdown-type format in Jupytext.